### PR TITLE
cli: rename --all to --all-remotes in branch list

### PR DESCRIPTION
**Pull request by Kintaro**
A follow up to #3406 

If applicable:
- [X] I have updated `CHANGELOG.md`
- [ ] I have updated the documentation (README.md, docs/, demos/)
- [ ] I have updated the config schema (cli/src/config-schema.json)
- [ ] I have added tests to cover my changes

### CODE REVIEW

File: cli/src/commands/branch.rs [hunk 1]
```diff
@@ -96,18 +96,24 @@ pub struct BranchDeleteArgs {
...
     /// Show all tracking and non-tracking remote branches including the ones
     /// whose targets are synchronized with the local branches.
-    #[arg(long, short, conflicts_with_all = ["names", "tracked"])]
+    #[arg(long, conflicts_with_all = ["names", "tracked"], hide = true)]
     all: bool,
 
+    /// Show all tracking and non-tracking remote branches including the ones
+    /// whose targets are synchronized with the local branches.
+    #[arg(long, short, conflicts_with_all = ["names", "tracked"])]
+    all_remotes: bool,
```
Comment (yuja) [inline cli/src/commands/branch.rs:108]: I think a hidden alias without deprecation (i.e. `alias = "all"`) is okay. What do you think?

Comment (Kintaro) [reply cli/src/commands/branch.rs:108]: Works for me. There's no direct need to deprecate the old style. Will update the PR

Comment (Kintaro) [reply cli/src/commands/branch.rs:102]: I have updated the PR to go with alias instead of deprecation

**Review by yuja — APPROVED** [review]
thanks

**Pull request merged**